scheduler/src/cook/mesos/scheduler.clj

```diff
@@ -522,158 +529,179 @@
                         (map job->usage)
                         (reduce (partial merge-with +)))))))
 
+(defn- category->pending-jobs->category->considerable-jobs
+  "Limit the pending jobs to considerable jobs based on usage and quota.
+   Further limit the considerable jobs to a maximum of num-considerable jobs."
+  [db pending-jobs user->quota user->usage num-considerable]
+  (log/debug "There are" (apply + (map count pending-jobs)) "pending jobs")
+  (log/debug "pending-jobs:" pending-jobs)
+  (let [category->considerable-jobs
+        (->> pending-jobs
+             (map (fn [[category jobs]]
```

Comment:
The interesting bit here is the filters on jobs. I propose putting that into a function called something like `filter-to-considerable` (or, you know, something way better) 

```clojure
(->> jobs
       ....)
```

Then this let block can become:

```clojure
(let [category->considerable-jobs (->> pending-jobs
                                  (map-vals filter-to-considerable))]
    ...)
```